src/backend/expungeservice/expunger/analyze.py

```diff
@@ -118,3 +134,166 @@ def __init__(self, name, dob, cases):
         self.name = name
         self.dob = dob
         self.cases = cases
+
+    def num_charges(self):
+        num = 0
+        for case in self.cases:
+            num += case.num_charges()
+        return num
+
+class ResultCode(enum.Enum):
+    INELIGIBLE = 'Ineligible'
+    ELIGIBLE = 'Eligible'
+    FURTHER_ANALYSIS = 'Further analysis needed'
+    EXAMINE = 'Examine'
+    NO_ACTION = 'Take no action'
+    OPEN_CASE = 'Open Case'
+
+"""
+A class for storing individual analysis steps
+
+Attributes:
+    result: A boolean that's the result of the check
+    check: A string that breifly describes the check
+    check_desc: A string that elaborates the check (optional)
+"""
+Check = collections.namedtuple('Check', 'result check check_desc',
+                               defaults=[None])
+
+class Result(object):
+    """
+    A class for storing analysis results
+
+    Attributes:
+        code: A ResultCode instance.
+        statute: A Statute instance that's associated with arrived result.
+        date: A datetime.date instance that specifies eligibility date.
+        checks: A list of Check instances that describes the analysis done.
+    """
+    def __init__(self, code=None, checks=None,
+                 statute=None, date=None):
+        self.code = code
+        self.checks = checks
+        self.statute = statute
```

Comment:
What is a result statute? Would the result statute not be the same as a charge statute?

Reply:
Hmm. I'm not sure about that. Let me check with Michael.

Reply:
So @michaelatcommunitylaw says the "expunge criteria statute" would be the same as the charge statute. I can remove this and probably add an assert / warning if they're not the same. Good catch!

Reply:
So we would have to maintain the statute after all as confirmed by Michael on the slack discussion.

Reply:
Cool, glad we clarified that!